simnode/common.go

```diff
@@ -50,18 +52,22 @@ type Args interface {
 // btcwalletArgs hold the different implementations
 type Node struct {
 	Args
-	handlers *rpc.NotificationHandlers
-	cmd      *exec.Cmd
-	client   *rpc.Client
-	pidFile  string
+	Handlers       *rpc.NotificationHandlers
+	Client         *rpc.Client
+	cmd            *exec.Cmd
+	pidFile        string
+	dataDir        string
+	maxConnRetries int
 }
 
 // NewNodeFromArgs starts a new node using the args provided, sets the handlers
 // and loggers. It does not start the node process, Start() should be called for that
-func NewNodeFromArgs(args Args, handlers *rpc.NotificationHandlers, w io.Writer) (*Node, error) {
+func NewNodeFromArgs(args Args, handlers *rpc.NotificationHandlers, w io.Writer, maxRetries int, dataDir string) (*Node, error) {
```

Comment:
I guess we're assuming `dataDir` already exists. We can update the docs mentioning this or create the directory if it doesn't exist.